packages/firebase/index.d.ts

```diff
@@ -1496,6 +1516,46 @@ declare namespace firebase.functions {
 }
 
 declare namespace firebase.auth {
+  /**
+   * A utility class to parse email action URLs.
+   */
+  class ActionCodeURL {
+    private constructor();
+    /**
+     * The API key of the email action link.
+     */
+    apiKey: string;
+    /**
+     * The action code of the email action link.
+     */
+    code: string;
+    /**
+     * The continue URL of the email action link. Null if not provided.
+     */
+    continueUrl: string | null;
+    /**
+     * The language code of the email action link. Null if not provided.
+     */
+    languageCode: string | null;
+    /**
+     * The email action type of the email action link. It returns from one
+     * of the types from {@link firebase.auth.ActionCodeInfo}.
+     */
+    operation: firebase.auth.ActionCodeInfo.Operation;
+    /**
+     * Parse the email action link string and returns an ActionCodeURL object
+     * if the link is valid, otherwise returns null.
+     *
+     * @param link The email action link string.
+     * @return The ActionCodeURL object, or null if the link is invalid.
+     */
+    static parseLink(link: string): firebase.auth.ActionCodeURL | null;
+    /**
+     * The tenant ID of the email action link. Null if the email action
+     * is from agent project.
```

Comment:
We don't use "agent" term externally.
Instead change to:
is from the parent project.

Reply:
done